arq/connections.py

```diff
@@ -150,19 +164,17 @@ async def enqueue_job(
             expires_ms = expires_ms or score - enqueue_time_ms + expires_extra_ms
 
             job = serialize_job(function, args, kwargs, _job_try, enqueue_time_ms, serializer=self.job_serializer)
-            tr = conn.multi_exec()
-            tr.psetex(job_key, expires_ms, job)
-            tr.zadd(_queue_name, score, job_id)
+            pipe.multi()
+            pipe.psetex(job_key, expires_ms, job)
+            pipe.zadd(_queue_name, {job_id: score})
             try:
-                await tr.execute()
-            except MultiExecError:
+                await pipe.execute()
+            except (ResponseError, WatchError):
```

Comment:
Not sure we need to catch `ResponseError` here too? This case is specifically for when the `watched` key get's set.

Reply:
I am pretty sure that at some point in my tests I got this error, now I can't reproduce it, removed it from except block.